app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -388,6 +390,21 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
         }
     }
 
+    private fun changeEngineMargins(swipeRefresh: View, inFullScreen: Boolean) {
+        swipeRefresh.apply {
+            val toolbarAndQASSize = resources.getDimension(R.dimen.toolbar_and_qab_height).toInt()
+            val toolbarSize = resources.getDimension(R.dimen.browser_toolbar_height).toInt()
+            (layoutParams as CoordinatorLayout.LayoutParams).apply {
+                setMargins(
```

Comment:
I find this logic hard to follow having known the context that it came from, so it might be harder for someone in the future to follow it as well.

Instead of inlining the logic in the place that it's being set, how about separate that logic outside, and then have a simple single setMargin with the final values?

[`DisplayToolbar`](https://github.com/mozilla-mobile/android-components/blob/master/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt#L300-L303) has an example of separating out the logic with a when block:

```kotlin
val (topMargin, bottomMargin) = when {
 // logic statements here.
}
setMargin(0, topMargin, 0, bottomMargin)
```

Just a suggestion, maybe this doesn't end up simplifying anything!

Reply:
Good call! I had never returned pairs like that, much cleaner!